packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/NotebookSessionController.java

```diff
@@ -250,29 +250,26 @@ ResponseEntity<NotebookSession> cloneNotebookSession(
 			currentUserService.get().getId(),
 			projectId
 		);
-		try {
-			final Optional<NotebookSession> session = sessionService.getAsset(id, permission);
-			if (session.isEmpty()) {
-				return ResponseEntity.notFound().build();
-			}
-			final NotebookSession newNotebookSession = sessionService.createAsset(
-				session.get().clone(),
-				projectId,
-				permission
-			);
-
-			final Project project = projectService
-				.getProject(projectId)
-				.orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("asset.not-found")));
-
-			projectAssetService.createProjectAsset(project, AssetType.NOTEBOOK_SESSION, newNotebookSession, permission);
+		final NotebookSession session = sessionService
+			.getAsset(id, permission)
+			.orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("asset.not-found")));
 
-			return ResponseEntity.status(HttpStatus.OK).body(newNotebookSession);
+		NotebookSession newNotebookSession;
+		try {
+			newNotebookSession = sessionService.createAsset(session.clone(), projectId, permission);
 		} catch (final Exception e) {
 			final String error = "Unable to clone notebook session";
 			log.error(error, e);
 			throw new ResponseStatusException(org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR, error);
 		}
+
+		final Project project = projectService
+			.getProject(projectId)
+			.orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("asset.not-found")));
```

Comment:
So I dunno how serious you are about handling exceptions the "right" way. We are fetching two types of resources here: project and notebook-session, and they are both throwing the generic "asset.not-found".

Which means if we do encounter this exception, it isn't obvious where the error is on the front-end, on the backend, because we repackage the error with no chaining, it is also not obvious where things have awry. So overall this isn't very useful.

Reply:
I agree, but because this is a rare occurence error. It's not too important which of the two is NOT_FOUND.

Nonetheless, you are correct and this is now fixed.